csrc/macros.h

```diff
@@ -12,6 +12,6 @@
 #include <bits/c++config.h>
 #endif
 
-#if defined(__GLIBCXX__) && __GLIBCXX__ >= 20230000
+#if defined(__GLIBCXX__) && __GLIBCXX__ >= 20230714
```

Comment:
I don't know from which version gcc start to support unordered_set with incomplete type. But `20230714` is the one locally on my disk and is tested to work.